src/bsp/src/stm32/src/CRC.cpp

```diff
@@ -0,0 +1,20 @@
+#include "CRC.h"
+#include "hal/hal.h"
+
+CRC::CRC() {
+    m_semaphore = xSemaphoreCreateBinary();
+
+    xSemaphoreGive(m_semaphore);
```

Comment:
No null check (though idk what you could do about it), maybe just add a todo to make static